frontend/packages/console-app/src/components/quick-starts/loader/QuickStartsLoader.tsx

```diff
@@ -0,0 +1,45 @@
+import * as React from 'react';
+import { QuickStart } from '../utils/quick-start-types';
+import { getQuickStarts } from '../utils/quick-start-utils';
+import QuickStartPermissionChecker from './QuickStartPermissionChecker';
+import { LoadingBox } from '@console/internal/components/utils';
+
+type QuickStartsLoaderProps = {
+  children: (quickStarts: QuickStart[]) => React.ReactNode;
+};
+
+const QuickStartsLoader: React.FC<QuickStartsLoaderProps> = ({ children }) => {
+  const [quickStarts, setQuickStarts] = React.useState<QuickStart[]>([]);
+  const [checkedQuickStarts, setCheckedQuickStarts] = React.useState<QuickStart[]>([]);
+  const permissionChecks = React.useRef<{ [name: string]: boolean }>({});
+
+  React.useEffect(() => {
+    setQuickStarts(getQuickStarts());
+  }, []);
+
+  return (
+    <>
+      {quickStarts.map((qs) => {
+        return (
+          <QuickStartPermissionChecker
+            key={qs.metadata.name}
+            quickStart={qs}
+            onPermissionCheck={(hasPermission) => {
+              if (hasPermission && !checkedQuickStarts.includes(qs)) {
+                setCheckedQuickStarts([...checkedQuickStarts, qs]);
+              }
+              permissionChecks.current[qs.metadata.name] = hasPermission;
+            }}
+          />
```

Comment:
Considering that `QuickStartPermissionChecker` has an effect whose dependency is the `onPermissionCheck` function. This should use `useCallback`.

Reply:
Done